From b84c97745f9f86ec6b0acd3d9173b171f1e07686 Mon Sep 17 00:00:00 2001
From: Cong Nguyen <cong.nguyen.zn@renesas.com>
Date: Tue, 15 Apr 2025 03:42:22 +0000
Subject: [PATCH 12/19] Add isu, fcpcs and vdpb nodes

Upstream-Status: Pending
---
 arch/arm64/boot/dts/renesas/r9a09g057.dtsi | 45 ++++++++++++++++++++++
 1 file changed, 45 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r9a09g057.dtsi b/arch/arm64/boot/dts/renesas/r9a09g057.dtsi
index d1609c525..f6ab322ce 100644
--- a/arch/arm64/boot/dts/renesas/r9a09g057.dtsi
+++ b/arch/arm64/boot/dts/renesas/r9a09g057.dtsi
@@ -1369,6 +1369,51 @@ du_out_dsi0: endpoint {
  			};
  		};
 
+		isu: isum@16450000 {
+				compatible = "renesas,isum";
+				reg = <0 0x16450000 0 0x01F4>;
+				interrupts =    <GIC_SPI 867 IRQ_TYPE_EDGE_RISING>,
+								<GIC_SPI 868 IRQ_TYPE_EDGE_RISING>,
+								<GIC_SPI 869 IRQ_TYPE_EDGE_RISING>,
+								<GIC_SPI 870 IRQ_TYPE_EDGE_RISING>,
+								<GIC_SPI 871 IRQ_TYPE_LEVEL_HIGH>,
+								<GIC_SPI 872 IRQ_TYPE_LEVEL_HIGH>,
+								<GIC_SPI 873 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&cpg CPG_MOD 230>,
+						<&cpg CPG_MOD 231>;
+				clock-names = "isu_aclk", "isu_pclk";
+				resets = <&cpg 213>,
+						<&cpg 214>;
+				reset-names = "aresetn", "presetn";
+				renesas,#ch = <0>;
+				renesas,#rpf = <1>;
+				renesas,#wpf = <1>;
+				renesas,has-rs;
+				power-domains = <&cpg>;
+		};
+
+		fcpcs: vcp4@16410000 {
+ 			compatible = "renesas,vcp4-fcpcs";
+ 			reg = <0 0x16410000 0 0x1000>;
+ 			renesas,#ch = <0>;
+ 		};
+
+ 		vdpb: vcp4@16400000 {
+ 			compatible = "renesas,vcp4-vdpb";
+ 			reg = <0 0x16400000 0 0x1000>,
+ 			      <0 0x16420200 0 0xE00>;
+ 			reg-names = "vlc", "ce";
+ 			interrupts = <GIC_SPI 887 IRQ_TYPE_LEVEL_HIGH>,
+ 				     <GIC_SPI 888 IRQ_TYPE_LEVEL_HIGH>;
+ 			clocks = <&cpg CPG_MOD 243>,
+ 				 <&cpg CPG_MOD 244>;
+ 			clock-names = "aclk", "pclk";
+ 			power-domains = <&cpg>;
+ 			resets = <&cpg 224>;
+ 			renesas,#ch = <0>;
+ 			renesas,#fcp_ch = <0>;
+ 		};
+
  		fcpvd: fcp@16470000 {
  			compatible = "renesas,r9a09g057-fcpvd",
  				     "renesas,fcpv";
-- 
2.25.1

